### PR TITLE
[UI-51] Time picker

### PR DESCRIPTION
**Pull request by stuniel**
#### Task

- **Issue or task referred:** UI-51
- **Feature or Bugfix:** Feature

I created a Time Picker component. It accepts and returns a time string in 'HH:mm' format and displays time in '12' and '24' format. Value can be selected from input or menu with time buttons.

#### Checklist

- [x] Required unit tests prepared
- [x] Documentation prepared
- [x] External licenses validated (ideally MIT)

### CODE REVIEW

File: packages/time-picker/src/TimeInput.js [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        onKeyDown={this.handleInputKeyDown}
+        right={this.buildControl()}
+        type='number'
+        suffix={suffix}
+        value={inputValue}
+        autoComplete='off'
+      />
+    )
+  }
+
```
Comment (maciek15) [inline packages/time-picker/src/TimeInput.js]: Clicking on the arrow icon does not open TimeMenu. Solution will be to set `pointer-events: none` to the Icon which is used here.

Comment (stuniel) [reply packages/time-picker/src/TimeInput.js]: :+1:

File: packages/time-picker/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+using `npm install @talixo/time-picker --save` or `yarn add @talixo/time-picker`.
+
+## Requirements
+
+Your package should additionally have some extra dependencies:
+
+- `prop-types: ^15.6.1`
+- `react: ^16.2.0`
+- `react-dom: ^16.2.0`
+- `@talixo/shared: ^0.1.0`
```
Comment (rangoo94) [inline packages/time-picker/README.md]: `moment` should be there

File: packages/time-picker/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+### TimePicker
+
+It allows any props which are allowed for `div`. Additionally, it handles some differently:
+
+Property name | Type        | Default | Description
+--------------|-------------|:-------:|--------------------------------
+className     | string      | n/a     | Additional class name passed to wrapper.
+onChange      | function    | n/a     | Event called after input value has been changed.
+hourFormat    | HourFormat  | `'HH'`  | Hour format.
```
Comment (rangoo94) [inline packages/time-picker/README.md]: Could you change `hourFormat` to 12 / 24?

File: packages/time-picker/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+value         | string      | n/a     | Time string in 'HH:mm' format passed to component.
+
+#### Types
+
+Type name      | Enum options
+---------------|---------------------------------------------------
+HourFormat     | 'HH', 'hh A'
+InputFormat    | 'HH', 'hh A', 'mm'
```
Comment (rangoo94) [inline packages/time-picker/README.md]: What is that input format?

File: packages/time-picker/src/TimeInput.js [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+    this.setState({ open: false })
+  }
+
+  /**
+   * Handles input focus.
+   * @param {string} value
+   *
+   */
+  handleFocus = () => {
+    // const { open } = this.state
```
Comment (rangoo94) [inline packages/time-picker/src/TimeInput.js]: Could you remove this comment?

File: packages/time-picker/src/TimeInput.js [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+  }
+
+  /**
+   * Handle keyboard events on input.
+   *
+   * @param {SyntheticEvent|Event} event
+   */
+  handleInputKeyDown = (event) => {
+    const { format } = this.props
+    // If format is differnet than 'hh A' - return
```
Comment (rangoo94) [inline packages/time-picker/src/TimeInput.js]: typo

File: packages/time-picker/README.md (hunk 2 above)
Comment (rangoo94) [reply packages/time-picker/README.md]: as well as:

    "prop-types": "^15.6.1",
    "react": "^16.2.0",
    "react-dom": "^16.2.0",
    "@talixo/button": "^0.1.0",
    "@talixo/icon": "^0.1.0",
    "@talixo/shared": "^0.1.0",
    "@talixo/text-input": "^0.1.0",
    "moment": "^2.22.1"

also, `moment` is duplicated in peerDependencies

**Comment by rangoo94** [comment]
Resolve conflict

**Pull request merged**